Bump angular to 9.0

oliverguenther opened the pull request:
- [x] Run migration to angular 9
- [x] Fixed ngx-gallery ivy incompatibility by using active fork
- [x] Fix TS errors with 3.7
- [x] Bump fullcalendar/angular to beta to fix ivy support
- [x] Check if we can move to Angular 9 suggested 'files' section in tsconfig. Due to our `require()`s, this requires us to specify more files there but will remove any unused files

- Currently, total saved bundle size in our production build appears to be `4MiB`. I would have expected more savings, but better than nothing. This is likely due to major factors being `moment` with all its locales and other libraries being pulled in that Ivy has no impact on

**Notable changes**

- `entryComponents` are no longer needed. This has the advantage of tree-shaking components that are never used in the code. However, we do have components such as `content-tabs` that are only used from Rails and were tree shaken with the current setup. You need to put components that are dynamically bootstrapped from Rails into the `src/app/global-dynamic-components.const.ts` definition to ensure they are manually imported and avoid tree shaking.

- We now use [TypeScript 3.7 with all its new features](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html) such as null-safe optional chaining `foo?.bar?.baz` and null/undefined check `foo ?? 'default'`.

- Angular 9 has the ngcc (ng compatibility compiler) to compile libraries into Ivy-supported formats for use. This will be done once, usually in the first `serve` run. Afterwards, compile time might increase somewhat but I found no too big difference on my machine.

- Abstract base classes / components that use `@Input` `@Output`, `@ViewChild` et al need to be marked a `@Directive()`, even though they are never put into any module `declarations`. This is simply an indiciation for the compiler to compile these annotations.

oliverguenther committed:
Bump angular to latest 8.x

oliverguenther committed:
Upgrade to angular/core 9.0

oliverguenther committed:
Bump cdk to 9.0

oliverguenther committed:
Remove deprecated `entryComponents`

oliverguenther committed:
Bump fullcalendar

oliverguenther committed:
Fix TS errors with 3.7

oliverguenther committed:
Use forked ngx-gallery to fix Ivy support

openprojectci commented:
<!--
  0 Errors
  18 Warnings: <a href="https://github.com/op...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          18 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/modals/export-modal/wp-table-export.modal.ts#L16">frontend/src/app/components/modals/export-modal/wp-table-export.modal.ts#L16</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/wp-relations/wp-relation-row/wp-relation-row.component.ts#L13">frontend/src/app/components/wp-relations/wp-relation-row/wp-relation-row.component.ts#L13</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/wp-relations/wp-relations-group/wp-relations-group.component.ts#L34">frontend/src/app/components/wp-relations/wp-relations-group/wp-relations-group.component.ts#L34</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/wp-table/configuration-modal/tabs/highlighting-tab.component.ts#L11">frontend/src/app/components/wp-table/configuration-modal/tabs/highlighting-tab.component.ts#L11</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/attachments/attachment-list/attachment-list.component.ts#L36">frontend/src/app/modules/attachments/attachment-list/attachment-list.component.ts#L36</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/attachments/attachments-upload/attachments-upload.component.ts#L37">frontend/src/app/modules/attachments/attachments-upload/attachments-upload.component.ts#L37</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/boards/board/board-list/board-list.component.ts#L57">frontend/src/app/modules/boards/board/board-list/board-list.component.ts#L57</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/boards/board/board.component.ts#L34">frontend/src/app/modules/boards/board/board.component.ts#L34</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/boards/board/inline-add/board-inline-add-autocompleter.component.ts#L53">frontend/src/app/modules/boards/board/inline-add/board-inline-add-autocompleter.component.ts#L53</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts#L29">frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts#L29</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/autocomplete/version-autocompleter.component.ts#L37">frontend/src/app/modules/common/autocomplete/version-autocompleter.component.ts#L37</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/ckeditor/op-ckeditor.component.ts#L42">frontend/src/app/modules/common/ckeditor/op-ckeditor.component.ts#L42</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/editable-toolbar-title/editable-toolbar-title.component.ts#L46">frontend/src/app/modules/common/editable-toolbar-title/editable-toolbar-title.component.ts#L46</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts#L48">frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts#L48</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/global_search/global-search-work-packages-entry.component.ts#L37">frontend/src/app/modules/global_search/global-search-work-packages-entry.component.ts#L37</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/global_search/global-search-work-packages.component.ts#L57">frontend/src/app/modules/global_search/global-search-work-packages.component.ts#L57</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/global_search/tabs/global-search-tabs.component.ts#L35">frontend/src/app/modules/global_search/tabs/global-search-tabs.component.ts#L35</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/global_search/title/global-search-title.component.ts#L38">frontend/src/app/modules/global_search/title/global-search-title.component.ts#L38</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

oliverguenther committed:
Remove unused files

openprojectci commented on frontend/src/app/modules/boards/boards-root/boards-root.component.ts:
```diff
@@ -3,7 +3,7 @@ import {BoardCacheService} from "core-app/modules/boards/board/board-cache.servi
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/efa080b533080e81ff35bb7f58518c16bf0b1632/frontend/src/app/modules/boards/boards-root/boards-root.component.ts#L3"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into bump/angular9

oliverguenther committed:
Fix injections in module

oliverguenther committed:
Bump uirouter

oliverguenther committed:
Bump more dependencies

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into bump/angular9

oliverguenther committed:
Bump danger

oliverguenther committed:
Remove unused spec

oliverguenther committed:
Bump parallel_tests

oliverguenther committed:
Re-add URIjs export

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into bump/angular9

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into bump/angular9

oliverguenther committed:
Fix EditForm callback not being destroyed in abstract base

oliverguenther committed:
Bump fullcalendar/integration

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into bump/angular9

openprojectci commented on frontend/src/app/modules/bim/bcf/bcf-wp-single-view/bcf-wp-single-view.component.ts:
```diff
@@ -1,7 +1,7 @@
-import { NgxGalleryOptions, NgxGalleryImage } from 'ngx-gallery';
+import {NgxGalleryImage, NgxGalleryOptions} from '@kolkov/ngx-gallery';
 import {HalLink} from "core-app/modules/hal/hal-link/hal-link";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/bim/bcf/bcf-wp-single-view/bcf-wp-single-view.component.ts#L6"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

oliverguenther committed:
Move DynamicBootstrapper to initializer to ensure import

Components are now tree shaken since entrycomponents no longer exists.
If they are never used in angular, they get removed. We have some
dynamically bootstrapped components from Rails that would get removed if
we do not explicitly import them.

openprojectci commented on frontend/src/app/ckeditor/ckeditor-augmented-textarea.component.ts:
```diff
@@ -26,23 +26,24 @@
 
 
+export const ckeditorAugmentedTextareaSelector = 'ckeditor-augmented-textarea';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/ckeditor/ckeditor-augmented-textarea.component.ts#L44"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/enterprise-banner/enterprise-banner-bootstrap.component.ts:
```diff
@@ -1,8 +1,10 @@
 import {I18nService} from "core-app/modules/common/i18n/i18n.service";
 
+export const enterpriseBannerSelector = 'enterprise-banner-bootstrap';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/enterprise-banner/enterprise-banner-bootstrap.component.ts#L5"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/homescreen/blocks/new-features.component.ts:
```diff
@@ -31,6 +31,8 @@ import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
 import {DomSanitizer} from "@angular/platform-browser";
 
+export const homescreenNewFeaturesBlockSelector = 'homescreen-new-features-block';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/homescreen/blocks/new-features.component.ts#L35"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/modals/confirm-form-submit/confirm-form-submit.directive.ts:
```diff
@@ -28,12 +28,13 @@
+import {Component, ElementRef, OnInit} from "@angular/core";
+
+export const confirmFormSubmitSelector = 'confirm-form-submit';
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/modals/confirm-form-submit/confirm-form-submit.directive.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/resizer/main-menu-resizer.component.ts:
```diff
@@ -26,16 +26,17 @@
 import {MainMenuToggleService} from "core-components/main-menu/main-menu-toggle.service";
 
+export const mainMenuResizerSelector = 'main-menu-resizer';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/resizer/main-menu-resizer.component.ts#L37"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-custom-actions/date-action/custom-date-action-admin.component.ts:
```diff
@@ -30,8 +30,11 @@ import {ApplicationRef, ChangeDetectorRef, Component, ElementRef, OnInit} from '
 
+
+export const customDateActionAdminSelector = 'custom-date-action-admin';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/wp-custom-actions/date-action/custom-date-action-admin.component.ts#L35"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-table/embedded/embedded-tables-macro.component.ts:
```diff
@@ -28,10 +28,11 @@
-import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
+
+export const wpEmbeddedTableEntrySelector = 'macro.embedded-table';
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/wp-table/embedded/embedded-tables-macro.component.ts#L33"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/components/wp-table/embedded/wp-embedded-table-entry.component.ts:
```diff
@@ -1,8 +1,9 @@
-import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
+
+export const wpTableEntrySelector = 'wp-embedded-table-entry';
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/components/wp-table/embedded/wp-embedded-table-entry.component.ts#L4"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/admin/types/type-form-configuration.component.ts:
```diff
@@ -30,8 +29,10 @@ export interface TypeGroup {
 }
 
+export const adminTypeFormConfigurationSelector = 'admin-type-form-configuration';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/admin/types/type-form-configuration.component.ts#L33"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/attachments/attachments.component.ts:
```diff
@@ -36,8 +36,10 @@ import {States} from 'core-components/states.service';
 import {filter, takeUntil} from 'rxjs/operators';
 
+export const attachmentsSelector = 'attachments';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/attachments/attachments.component.ts#L40"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/bim/bcf/bcf-buttons/bcf-export-button.component.ts:
```diff
@@ -26,17 +26,15 @@
 import {UrlParamsHelperService} from "core-components/wp-query/url-params-helper";
 import {StateService} from "@uirouter/core";
-import {WorkPackageStaticQueriesService} from "core-components/wp-query-select/wp-static-queries.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/bim/bcf/bcf-buttons/bcf-export-button.component.ts#L38"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/bim/bcf/bcf-buttons/bcf-import-button.component.ts:
```diff
@@ -28,7 +28,6 @@
-import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
 import {CurrentProjectService} from "core-components/projects/current-project.service";
 import {BcfPathHelperService} from "core-app/modules/bim/bcf/helper/bcf-path-helper.service";
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/bim/bcf/bcf-buttons/bcf-import-button.component.ts#L33"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/boards/boards-sidebar/boards-menu.component.ts:
```diff
@@ -3,14 +3,15 @@ import {Observable} from "rxjs";
 import {CurrentProjectService} from "core-components/projects/current-project.service";
 
+export const boardsMenuSelector = 'boards-menu';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/boards/boards-sidebar/boards-menu.component.ts#L12"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/autocomplete/autocomplete-select-decoration.component.ts:
```diff
@@ -26,12 +26,13 @@
 type SelectItem = { label:string, value:string, selected?:boolean };
 
+export const autocompleteSelectDecorationSelector = 'autocomplete-select-decoration';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/autocomplete/autocomplete-select-decoration.component.ts#L35"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/autocomplete/create-autocompleter.component.ts:
```diff
@@ -27,19 +27,18 @@
+  name:string;
+  $href:string|null;
+}
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/autocomplete/create-autocompleter.component.ts#L42"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/autocomplete/user-autocompleter.component.ts:
```diff
@@ -39,6 +39,8 @@ import {NgSelectComponent} from "@ng-select/ng-select";
 import {HalResource} from "core-app/modules/hal/resources/hal-resource";
 
+export const usersAutocompleterSelector = 'user-autocompleter';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/autocomplete/user-autocompleter.component.ts#L43"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/collapsible-section/collapsible-section.component.ts:
```diff
@@ -28,7 +28,6 @@
-import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
 
 export const collapsibleSectionAugmentSelector = 'collapsible-section-augment';
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/collapsible-section/collapsible-section.component.ts#L33"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/colors/colors-autocompleter.component.ts:
```diff
@@ -27,10 +27,11 @@
 import {I18nService} from "core-app/modules/common/i18n/i18n.service";
 
+export const colorsAutocompleterSelector = 'colors-autocompleter';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/colors/colors-autocompleter.component.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/copy-to-clipboard/copy-to-clipboard.directive.ts:
```diff
@@ -27,16 +27,16 @@
-import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
+
+export const copyToClipboardSelector = 'copy-to-clipboard';
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/copy-to-clipboard/copy-to-clipboard.directive.ts#L36"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/hide-section/add-section-dropdown/add-section-dropdown.component.ts:
```diff
@@ -30,11 +30,12 @@ import {I18nService} from "core-app/modules/common/i18n/i18n.service";
 import {AngularTrackingHelpers} from "core-components/angular/tracking-functions";
 
+export const addSectionDropdownSelector = 'add-section-dropdown';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/hide-section/add-section-dropdown/add-section-dropdown.component.ts#L36"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/hide-section/hide-section-link/hide-section-link.component.ts:
```diff
@@ -30,8 +30,10 @@ import {Component, ElementRef, OnInit} from "@angular/core";
 import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
 
+export const hideSectionLinkSelector = 'hide-section-link';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/hide-section/hide-section-link/hide-section-link.component.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/hide-section/show-section-dropdown.component.ts:
```diff
@@ -30,8 +30,10 @@ import {HideSectionService} from "./hide-section.service";
 import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
 
+export const showSectionDropdownSelector = 'show-section-dropdown';
+
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/hide-section/show-section-dropdown.component.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/persistent-toggle/persistent-toggle.component.ts:
```diff
@@ -27,10 +27,11 @@
-import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
+
+export const persistentToggleSelector = 'persistent-toggle';
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/persistent-toggle/persistent-toggle.component.ts#L32"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/common/remote-field-updater/remote-field-updater.component.ts:
```diff
@@ -29,10 +29,11 @@
-import {DynamicBootstrapper} from 'core-app/globals/dynamic-bootstrapper';
+
+export const remoteFieldUpdaterSelector = 'remote-field-updater';
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/common/remote-field-updater/remote-field-updater.component.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

openprojectci commented on frontend/src/app/modules/router/base/application-base.component.ts:
```diff
@@ -27,7 +27,6 @@
-import {DynamicBootstrapper} from "core-app/globals/dynamic-bootstrapper";
 
 export const appBaseSelector = 'openproject-base';
 
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/be01da6b0fff41b1a83c29ab070c319c13aa844c/frontend/src/app/modules/router/base/application-base.component.ts#L32"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>
